src/lib/DeBridge/parser.ts

```diff
@@ -0,0 +1,61 @@
+import { ethers } from "ethers";
+import { ACTION_ENUM } from "../../enums";
+import { ProtocolHelper } from "../../helpers";
+import { IBridgeInAction, IBridgeOutAction, ITransaction, ITransactionAction, ITransactionLog } from "../../types";
+import { CONTRACT_ENUM, EVENT_ENUM, contracts } from "./contracts";
+
+export class DeBridgeParser {
+  public static async parseTransaction(transaction: ITransaction): Promise<ITransactionAction[]> {
+    for (const log of transaction.logs) {
+      const topic = log.topics[0].toLowerCase();
+      
+      if (topic === EVENT_ENUM.SENT.toLowerCase()) {
+        return [this.parseSentEvent(log, transaction)];
+      }
+
+      if (topic === EVENT_ENUM.CLAIMED.toLowerCase()) {
+        return [this.parseClaimedEvent(log, transaction)];
+      }
+    }
+
+    throw new Error("No supported events found in transaction");
+  }
+
+  private static parseSentEvent(log: ITransactionLog, transaction: ITransaction): IBridgeOutAction {
+    const parsedLog = ProtocolHelper.parseLog(
+      log,
+      contracts[CONTRACT_ENUM.DEBRIDGE_GATE].events[EVENT_ENUM.SENT]
+    );
+
+    return {
+      type: ACTION_ENUM.BRIDGE_OUT,
+      fromChain: transaction.chainId,
+      toChain: Number(parsedLog.args.chainIdTo),
+      fromToken: parsedLog.args.feeParams.isNativeToken ? ethers.ZeroAddress : transaction.to.toLowerCase(),
```

Comment:
are you sure about this?
what if the its a token being bridged out but fees being  paid in native token?
Please confirm this scenario!

also no need to lowercase token addresses. return them as it is
anywhere any address requires no need of any conversion, its being handled outside

Reply:
Also in case its not native you are saving the fromToken as the "to" field of the transaction?
to will be the contract address